### ISSUE TITLE
Attaching a new serial interface reboots the UX Module

### ISSUE THREAD

**Issue by feilipu** [feilipu/ux_module#11]
Attaching a new terminal to `cat` code up to the Z80 causes the UX Module to reboot.

We need this function to upload code to the Z80 without needing to copy/pasta it into the initial open terminal.

Probably this is caused by some kind of Linux side DTR toggling when opening a `/dev/ttyUSB0` connection, but it might be something else.

**Comment by Wuerfel21** [comment]
IDK about Linux, but on Windows there is a setting deep in the COM port config dialog for this

**Comment by feilipu** [comment]
Will look into Windows, to see whether same issue occurs.

How does text upload work in Windows best? Then I can copy method to test.

Linux side there are many `stty` configuration options to try, but they are cryptic, not generally perfect, and till now I've not cracked the code.

Need to check into what exactly is happening first though.

**Comment by feilipu** [comment]
This is a known problem in Linux (of all varieties), which the Arduino world has been struggling with for years.
https://github.com/pyserial/pyserial/issues/124
https://stackoverflow.com/questions/15460865/disable-dtr-in-pyserial-from-code
https://raspberrypi.stackexchange.com/questions/9695/disable-dtr-on-ttyusb0
https://playground.arduino.cc/Main/DisablingAutoResetOnSerialConnection/
Effectively, you can't use an Arduino remotely, because disconnecting / reconnecting the serial port will reboot the machine.

In summary the required `stty -cdtrdsr` option is not implemented, and the `stty -F /dev/ttyUSB0 -hupcl` option doesn't really work. Not sure how this can be fixed at the moment.

Even more annoyingly just using `stty -F /dev/ttyUSB0 -a` to read the port causes it to reboot the UX Module.

**Comment by feilipu** [comment]
Dug as deep as I can.
Wrote this based on references, and experimented with variations on the theme.
Nothing works.

```python
import os
import sys
import serial
# import termios

from time import sleep

path = '/dev/ttyUSB0'
baud = 115200

# Disable reset after hangup
# with open(path) as f:
#    attrs = termios.tcgetattr(f)
#    attrs[2] = attrs[2] & ~termios.HUPCL & termios.CLOCAL
#    termios.tcsetattr(f, termios.TCSAFLUSH, attrs)

os.system("stty -F /dev/ttyUSB0 clocal -hupcl")

ser = serial.Serial(path, baud, dsrdtr=False)

# print(ser)

for line in sys.stdin:

    for ch in line:
        ser.write( ch );

    ser.write('\r');
    ser.flush();
    sleep(0.1);

ser.close()
sys.exit()
```

**Comment by feilipu** [comment]
Closed with [documentation](https://github.com/feilipu/ux_module/commit/3b8ac09a50e496cf18c369c16778935e791145ac).
